config/controller.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2019 Samaritan Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/samaritan-proxy/sash/logger"
+)
+
+const (
+	NamespaceService       = "service"
+	TypeServiceProxyConfig = "proxy-config"
+	TypeServiceDependence  = "dependence"
+)
+
+var InterestedNSAndType = map[string][]string{
+	NamespaceService: {TypeServiceProxyConfig, TypeServiceDependence},
+}
+
+// Controller is used to store configuration information.
+type Controller struct {
+	sync.Mutex
+	store    Store
+	updateCh chan struct{}
+
+	cache    atomic.Value // *Config
+	interval time.Duration
+
+	evtHdls []EventHandler
+
+	stop chan struct{}
+	wg   sync.WaitGroup
+
+	initFinish bool
+}
+
+// NewController return a new Controller.
+func NewController(store Store, interval time.Duration) *Controller {
```

Comment:
It's best to pass the interval as an option，like `NewController(store Store, options ...controllerOption)` .